### PR TITLE
6168 pitch and speed focus fix

### LINKED ISSUE #6168: Pitch and Speed dialog stays hidden when accessed via overflow button
### Bug description

Pitch and Speed dialog stays behind main window when accessed via overflow button if it has already been opened and is in the background.

### Steps to reproduce

1. Invoke the dialog
2. Click somewhere else - the dialog is now hidden
3. Invoke the dialog using the overflow button

### Expected behavior

The dialog is brought to the fore.

### Actual behavior

The dialog stays hidden.

### Audacity Version

current master / alpha / nightly build

### Operating system

macOS

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by saintmatthieu**
Resolves: #6168, #6170

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by saintmatthieu** [comment]
@DavidBailes this actually depends on your fix, so to reduce overhead I included it in this PR.

**Review by crsib — COMMENTED** [review]
I don't quite understand what is goal of this PR, except for parenting.

I think what you a looking for is `wxDialog::Raise`, but probably I am wrong.

**Comment by DavidBailes** [comment]
> I think what you a looking for is `wxDialog::Raise`, but probably I am wrong.

OnNextWindow() and OnPrevWindow() both use Raise(). There's also a mac/Linux only SetFocus().
If you haven't already tried using Raise() might be worth doing so to avoid the hide/show workaround.

**Comment by saintmatthieu** [comment]
Yes, looks good! Coming up ...

File: src/tracks/playabletrack/wavetrack/ui/PitchAndSpeedDialog.cpp [hunk 1]
```diff
@@ -269,13 +269,8 @@ PitchAndSpeedDialog& PitchAndSpeedDialog::SetFocus(
...
    if (item)
       item->SetFocus();
-#ifndef _WIN32
-   // On macOS, just calling `SetFocus` isn't enough, but on Windows, changing
-   // visibility to false and then back to true is ugly, so don't do this there.
-   // TODO: Linux ?
-   Show(false);
-#endif
-   Show(true);
+   wxDialog::Show(true);
```
Comment (crsib) [inline src/tracks/playabletrack/wavetrack/ui/PitchAndSpeedDialog.cpp:272]: But why do we still need both Hide/Show cycle and Raise? I expected that Raise would be sufficient alone

Comment (saintmatthieu) [reply src/tracks/playabletrack/wavetrack/ui/PitchAndSpeedDialog.cpp:272]: Hide/Show cycle? We are only calling `Show(true)`. If that's what you meant: maybe, but does it hurt?

Comment (crsib) [reply src/tracks/playabletrack/wavetrack/ui/PitchAndSpeedDialog.cpp:272]: Hmm, I misread the diff, sorry

**Review by crsib — APPROVED** [review]

**Pull request merged**